### PR TITLE
chore: model test refactor, models improvements, mock cleanup

### LINKED ISSUE #143: Research testing utilities and data structure and dependencies
There are currently several test suites which would benefit from sharing test-specific utility functions, mocks and other.

1. #157 
2. Identify generic helper functions which may be shared.
3. Identify fixtures and other data which should be moved to examples.

This issue should be done once a better understanding of the test suites is gained - after #145

### PR DESCRIPTION
**Pull request by Tlacenka**
Closes #143, #276, #254 

This is the second part of models refactor:
- I finished cleaning up models tests.
- I migrated mocks to `testing-utils`.
- I fixed models - `plugins` cannot be empty, missing reference correctly throws for empty `audits` array, `packageName` and `version` is correctly required.
- Require slug for `project` and `organization` in upload configuration (otherwise the string is rejected by GraphQL).

### CODE REVIEW

File: packages/models/src/lib/plugin-config.unit.test.ts [hunk 1]
```diff
@@ -1,101 +1,109 @@
...
 
-  it('should throw if plugin audits contain invalid slugs', () => {
-    const invalidAuditRef = '-invalid-audit-slug';
-    const pluginConfig = config().plugins[0];
-    pluginConfig.audits[0].slug = invalidAuditRef;
-
-    expect(() => pluginConfigSchema.parse(pluginConfig)).toThrow(
-      `slug has to follow the patter`,
-    );
+  it('should accept a plugin configuration with no audits or groups', () => {
```
Comment (matejchalk) [inline packages/models/src/lib/plugin-config.unit.test.ts:43]: It's questionable if this is correct, a plugin without audits doesn't measure anything (only groups are optional).

Out of scope of this PR, though.

File: packages/models/src/lib/category-config.unit.test.ts [hunk 2]
```diff
@@ -1,29 +1,252 @@
...
+              type: 'audit',
+              weight: 1,
+            },
+          ],
+        },
+      ] satisfies CategoryConfig[]),
+    ).not.toThrow();
+  });
+
+  it('should throw for an empty category array', () => {
```
Comment (matejchalk) [inline packages/models/src/lib/category-config.unit.test.ts:214]: Perhaps categories should be optional, i.e. could be missing or empty array? :thinking: Motivation would be to simplify onboarding.

Out of scope of this PR.

Comment (matejchalk) [inline packages/models/src/lib/category-config.unit.test.ts:214]: Perhaps categories should be optional, i.e. could be missing or empty array? :thinking: Motivation would be to simplify onboarding.

Out of scope of this PR.

File: packages/models/src/lib/upload-config.ts [hunk 3]
```diff
@@ -1,18 +1,14 @@
...
       'API key with write access to portal (use `process.env` for security)',
   }),
-  organization: z.string({
-    description: 'Organization in code versioning system',
-  }),
-  project: z.string({
-    description: 'Project in code versioning system',
-  }),
+  organization: slugSchema('Organization in code versioning system'),
+  project: slugSchema('Project in code versioning system'),
```
Comment (matejchalk) [inline packages/models/src/lib/upload-config.ts:11]: I notice the descriptions aren't very accurate :sweat_smile: 

```suggestion
  organization: slugSchema('Organization slug from Code PushUp portal'),
  project: slugSchema('Project slug from Code PushUp portal'),
```

**Review by matejchalk — COMMENTED** [review]
Looks good :+1:

File: packages/models/src/lib/plugin-config.unit.test.ts (hunk 1 above)
Comment (Tlacenka) [reply packages/models/src/lib/plugin-config.unit.test.ts:43]: #406 created

File: packages/models/src/lib/category-config.unit.test.ts (hunk 2 above)
Comment (Tlacenka) [reply packages/models/src/lib/category-config.unit.test.ts:214]: Raised in #407

File: packages/models/src/lib/upload-config.ts (hunk 3 above)
Comment (Tlacenka) [reply packages/models/src/lib/upload-config.ts]: Good point, fixed.

**Review by matejchalk — APPROVED** [review]

**Pull request merged**